include/extractor/guidance/intersection.hpp

```diff
@@ -204,6 +212,13 @@ template <typename Self> struct EnableIntersectionOps
     // Returns the number of roads we can not enter at this intersection, respectively.
     auto countNonEnterable() const { return self()->size() - self()->countEnterable(); }
 
+    // check if all roads between begin and end allow entry
+    template <typename iteratorT>
+    bool hasAllValidEntries(const iteratorT begin, const iteratorT end) const
+    {
+        return all_of(begin, end, [](const ConnectedRoad &road) { return road.entry_allowed; });
+    }
```

Comment:
`std::all_of` and not `all_of` - the reason the unqualified `all_of` works is argument-dependant lookup (ADL, great to know, read up a bit about it): if the iterators are in the std namespace `all_of` is searched in the std namespace, too.

Check `std::all_of`s docs: it requires an `InputIt` (make yourself comfortable with the different iterator [concepts](http://en.cppreference.com/w/cpp/iterator)), which basically means you can run over it once. Because you are using `std::all_of` your iterators now also have to be (at least) `InputIterators`. You _could_ assert that at compile time via

    static_assert(std::is_base_of<std::input_iterator_tag, typename std::iterator_traits<Iter>::iterator_category>::value, "")

but what you usually do is to make the template's type just InputIt, as in

    template <typename InputIt> void fn(InputIt first, InputIt last);

(C++ Concepts will change this in the future and enforce the static assert automatically)

Finally, don't hard code `ConnectedRoad` as parameter here: the reason we're having the `EnableOps` CRTP struct is because you don't know if it's a `ConnectedRoad` or a `ViewData`. Make it generic.